smartsim/settings/launch_settings.py

```diff
@@ -165,6 +174,12 @@ def env_vars(self, value: dict[str, str | None]) -> None:
 
         :param value: The new environment mapping
         """
+        if not (
+            isinstance(value, dict)
+            and all(isinstance(key, str) for key, val in value.items())
```

Comment:
Similar 'should we runtime type check the values of `value`' comment here as with `BatchSettings.env_vars`